src/flags/coord_flag_env_handler.cpp

```diff
@@ -22,61 +24,91 @@
 #include <range/v3/range/conversion.hpp>
 #include <range/v3/view/drop_while.hpp>
 #include <range/v3/view/take_while.hpp>
+#include <range/v3/view/transform.hpp>
 
 namespace memgraph::flags {
 
-CoordinationSetup::CoordinationSetup(int management_port, int coordinator_port, uint32_t coordinator_id)
-    : management_port(management_port), coordinator_port(coordinator_port), coordinator_id(coordinator_id) {}
+CoordinationSetup::CoordinationSetup(int management_port, int coordinator_port, uint32_t coordinator_id,
+                                     std::string nuraft_log_file)
+    : management_port(management_port),
+      coordinator_port(coordinator_port),
+      coordinator_id(coordinator_id),
+      nuraft_log_file(std::move(nuraft_log_file)) {}
 
 std::string CoordinationSetup::ToString() {
-  return fmt::format("management port: {}, coordinator port {}, coordinator id", management_port, coordinator_port,
-                     coordinator_id);
+  return fmt::format("management port: {}, coordinator port: {}, coordinator id: {}, nuraft_log_file: {}",
+                     management_port, coordinator_port, coordinator_id, nuraft_log_file);
 }
 
+// TODO: (andi) This will change if we add management server to coordinators.
 [[nodiscard]] auto CoordinationSetup::IsCoordinatorManaged() const -> bool { return management_port != 0; }
 
 auto CoordinationSetupInstance() -> CoordinationSetup & {
-  static auto instance = CoordinationSetup{0, 0, 0};
+  static auto instance = CoordinationSetup{};
   return instance;
 }
 
 void SetFinalCoordinationSetup() {
 #ifdef MG_ENTERPRISE
-  auto const *maybe_management_port = std::getenv(kMgManagementPort);
-  auto const *maybe_coordinator_port = std::getenv(kMgCoordinatorPort);
-  auto const *maybe_coordinator_id = std::getenv(kMgCoordinatorId);
 
-  bool const are_envs_set = maybe_management_port || maybe_coordinator_port || maybe_coordinator_id;
-  bool const are_flags_set = FLAGS_management_port || FLAGS_coordinator_port || FLAGS_coordinator_id;
+  std::vector<char const *> const maybe_coord_envs{std::getenv(kMgManagementPort), std::getenv(kMgCoordinatorPort),
+                                                   std::getenv(kMgCoordinatorId), std::getenv(kMgNuRaftLogFile)};
 
-  if (are_envs_set && are_flags_set) {
-    spdlog::trace(
-        "Ignoring coordinator setup(management_port, coordinator_port and coordinator_id) sent via flags as there is "
-        "input in environment variables");
-  }
+  bool const any_envs_set = std::ranges::any_of(maybe_coord_envs, [](char const *env) { return env != nullptr; });
+
+  std::array<std::any, 4> maybe_coord_flags{FLAGS_management_port, FLAGS_coordinator_port, FLAGS_coordinator_id,
+                                            FLAGS_nuraft_log_file};
 
-  auto const canonicalize_string = [](auto &&rng) {
-    auto const is_space = [](auto c) { return c == ' '; };
+  auto const is_flag_set = [](auto const &flag) {
```

Comment:
Don't use std::any, you can just compare with default initialized values.

```cpp
auto const is_flag_set = []<typename T>(T const &flag) { return flag != T{}; };
```

Reply:
But I cannot use then std::array? Or which collection can I use then, can you expand?

Reply:
solved on the call